### ISSUE TITLE
pachd nodes crash when deleting repo with pipeline watching

### ISSUE THREAD

**Issue by erikreppel** [pachyderm/pachyderm#435]
Can be recreated by following the fruit stand example then 
```$ pachctl delete-repo filter```
wait about 5 seconds and the mount will complain and `kubectl get all` will show errors on the pachd nodes.

The crash still occurs if you delete the pipeline watching that repo and then delete the repo
```
$ pachctl delete-pipeline filter
$ pachctl delete-repo filter
```

It also looks like the pachd nodes never recover after failing, they get to this point 
```
NAME                   DESIRED      CURRENT       AGE
etcd                   1            1             6m
pachd                  2            2             6m
rethink                1            1             6m
NAME                   CLUSTER-IP   EXTERNAL-IP   PORT(S)                        AGE
etcd                   10.0.0.111   <none>        2379/TCP,2380/TCP              6m
kubernetes             10.0.0.1     <none>        443/TCP                        35d
pachd                  10.0.0.237   nodes         650/TCP                        6m
rethink                10.0.0.172   nodes         8080/TCP,28015/TCP,29015/TCP   6m
NAME                   READY        STATUS        RESTARTS                       AGE
etcd-7wilb             1/1          Running       0                              6m
k8s-etcd-127.0.0.1     1/1          Running       0                              35d
k8s-master-127.0.0.1   4/4          Running       0                              18d
k8s-proxy-127.0.0.1    1/1          Running       0                              18d
pachd-l0lhy            0/1          Running       5                              6m
pachd-p75ud            0/1          Running       5                              6m
rethink-n1ifm          1/1          Running       0                              6m

```
but then immediately crash again.

**Comment by derekchiang** [comment]
Fixed in #437 (which is wrongly named fix-436 lol).

**Comment by derekchiang** [comment]
Actually this is not completely fixed.  You can still delete a repo when a running pipeline is outputting to it.  That shouldn't be allowed.

**Comment by derekchiang** [comment]
Fixed in #440.  Now you will should be able to see that your pipeline errored if you removed your pipeline's output repo, by using `list-pipeline` or `inspect-pipeline`
